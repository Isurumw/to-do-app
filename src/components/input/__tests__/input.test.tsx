import 'react-native';
import React from 'react';

import Input from '../input';
import {render, fireEvent} from 'react-native-testing-library';

describe('check the behaviour of the input component', () => {
  it('the input component renders correctly', () => {
    const component = render(<Input label="Email" type="email" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('the input component renders correctly, when it is password', () => {
    const component = render(<Input label="Password" type="password" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('press the eye icon to see or hide the password', () => {
    const mockHandleChangeText: jest.Mock<void> = jest.fn();
    const password = 'todo@123';

    const component = render(
      <Input
        label="Password"
        type="password"
        inputProps={{onChangeText: mockHandleChangeText}}
      />,
    );

    const textInput = component.getByTestId('input_test');
    const btnEye = component.getByTestId('button_eye');

    fireEvent.changeText(textInput, password);
    expect(mockHandleChangeText.mock.calls).toEqual([[password]]);

    fireEvent.press(btnEye);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
