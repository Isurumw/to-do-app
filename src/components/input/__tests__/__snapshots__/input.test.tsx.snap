// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the behaviour of the input component press the eye icon to see or hide the password 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#5C677C",
        "fontFamily": "DMSans-Medium",
        "fontSize": 14,
        "letterSpacing": -0.04,
        "lineHeight": 21,
      }
    }
  >
    Password
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 1,
          "marginTop": 4,
          "paddingHorizontal": 16,
          "paddingVertical": 15,
        },
        Object {
          "borderColor": "#D5D8DF",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={false}
      onBlur={[Function]}
      onChangeText={
        [MockFunction] {
          "calls": Array [
            Array [
              "todo@123",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      onFocus={[Function]}
      secureTextEntry={false}
      selectionColor="#7C2AF7"
      style={
        Object {
          "color": "#2E3C57",
          "flex": 1,
          "fontFamily": "DMSans-Regular",
          "fontSize": 15,
          "letterSpacing": -0.13,
        }
      }
      testID="input_test"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button_eye"
    >
      <SvgMock
        fill="#2E046C"
        height={20}
        width={20}
      />
    </View>
  </View>
</View>
`;

exports[`check the behaviour of the input component the input component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#5C677C",
        "fontFamily": "DMSans-Medium",
        "fontSize": 14,
        "letterSpacing": -0.04,
        "lineHeight": 21,
      }
    }
  >
    Email
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 1,
          "marginTop": 4,
          "paddingHorizontal": 16,
          "paddingVertical": 15,
        },
        Object {
          "borderColor": "#D5D8DF",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={false}
      onBlur={[Function]}
      onFocus={[Function]}
      secureTextEntry={false}
      selectionColor="#7C2AF7"
      style={
        Object {
          "color": "#2E3C57",
          "flex": 1,
          "fontFamily": "DMSans-Regular",
          "fontSize": 15,
          "letterSpacing": -0.13,
        }
      }
      testID="input_test"
    />
  </View>
</View>
`;

exports[`check the behaviour of the input component the input component renders correctly, when it is password 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#5C677C",
        "fontFamily": "DMSans-Medium",
        "fontSize": 14,
        "letterSpacing": -0.04,
        "lineHeight": 21,
      }
    }
  >
    Password
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 1,
          "marginTop": 4,
          "paddingHorizontal": 16,
          "paddingVertical": 15,
        },
        Object {
          "borderColor": "#D5D8DF",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={false}
      onBlur={[Function]}
      onFocus={[Function]}
      secureTextEntry={true}
      selectionColor="#7C2AF7"
      style={
        Object {
          "color": "#2E3C57",
          "flex": 1,
          "fontFamily": "DMSans-Regular",
          "fontSize": 15,
          "letterSpacing": -0.13,
        }
      }
      testID="input_test"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button_eye"
    >
      <SvgMock
        fill="#2E046C"
        height={20}
        width={20}
      />
    </View>
  </View>
</View>
`;
